ddtrace/internal/globaltracer.go

```diff
@@ -6,35 +6,33 @@
 package internal // import "gopkg.in/DataDog/dd-trace-go.v1/ddtrace/internal"
 
 import (
-	"sync"
+	"sync/atomic"
 
 	"gopkg.in/DataDog/dd-trace-go.v1/ddtrace"
 )
 
 var (
-	mu           sync.RWMutex   // guards globalTracer
-	globalTracer ddtrace.Tracer = &NoopTracer{}
+	// globalTracer stores the current tracer as *ddtrace.Tracer (pointer to interface). The
+	// atomic.Value type requires types to be consistent, which requires using *ddtrace.Tracer.
+	globalTracer atomic.Value
 )
 
+func init() {
+	var tracer ddtrace.Tracer = &NoopTracer{}
+	globalTracer.Store(&tracer)
```

Comment:
I might be saying something stupid here, but I've never used `atomic.Value` before. Can't we just do the below?
```suggestion
	globalTracer.Store(ddtrace.Tracer(&NoopTracer{}))
```
Does it have to be a pointer to an interface? Seems awkward but maybe it's just my lack of experience with this structure.